File: A1/C_files/Assignment1.c
/* Assignment1.c - driver code for the
                   lexical analyzer */
#include <stdio.h>
#include "LexicalAnalyzer.h"

/******************************************************/
/* main driver */
int main(int argc, char *argv[])
{
  FILE *in_fp;
  Token nextToken;

/* Open the input data file and process its contents */
  if  ((in_fp = fopen(argv[1], "r")) == NULL)
    printf("ERROR - cannot open input file \n");
  else
  {
    initialize(in_fp);
    do
    {
      nextToken = getToken();
      printf("Next token is: %d, Next lexeme is %s\n",
             nextToken.tokenCode, nextToken.lexemeString);
    } while (nextToken.tokenCode != EOF);
  }
}

File: A1/java_files/Assignment1.java
import java.util.*;
import java.io.*;

public class Assignment1{
  public static void main(String[] args){    
    LexicalAnalyzer myLA = new LexicalAnalyzer(args[0]);
    Token token;
    do{
      token = myLA.getToken();
      System.out.println(String.format("Next token is %d, Next lexeme is %s", token.getTokenCode(), token.getLexemeString() ));
    } while(token.getTokenCode()!=-1);
  }
}